src/libcollections/slice.rs

```diff
@@ -1031,9 +1031,21 @@ pub trait SliceConcatExt<T: ?Sized> {
     /// # Examples
     ///
     /// ```
+    /// assert_eq!(["hello", "world"].join(" "), "hello world");
+    /// ```
+    #[stable(feature = "rust1", since = "1.0.0")]
+    fn join(&self, sep: &T) -> Self::Output;
+
+    /// Flattens a slice of `T` into a single value `Self::Output`, placing a
+    /// given separator between each.
+    ///
+    /// # Examples
+    ///
+    /// ```
     /// assert_eq!(["hello", "world"].connect(" "), "hello world");
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
+    #[deprecated(since = "1.3.0", reason = "renamed to join")]
     fn connect(&self, sep: &T) -> Self::Output;
```

Comment:
Having a seperate "rename" and "fallout" commit seems reasonable.

On Sat, Jul 11, 2015 at 1:42 PM, Wesley Wiser <notifications@github.com>
wrote:

> In src/libcollections/slice.rs
> <https://github.com/rust-lang/rust/pull/26957#discussion_r34416058>:
>
> >      /// assert_eq!(["hello", "world"].connect(" "), "hello world");
> >      /// ```
> >      #[stable(feature = "rust1", since = "1.0.0")]
> > +    #[deprecated(since = "1.3.0", reason = "renamed to join")]
> >      fn connect(&self, sep: &T) -> Self::Output;
>
> Ok. I'll change connect to default to join which should address the
> remaining nits. I'll push that change up in a minute. Should I squash
> everything together or leave the rename and connect-to-join changes in
> two separate commits?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rust-lang/rust/pull/26957/files#r34416058>.
>